test/types/instance.test-d.ts

```diff
@@ -130,3 +136,13 @@ expectType<string>(server.printRoutes({ includeHooks: true, commonPrefix: false,
 expectType<string>(server.printRoutes({ includeMeta: ['key1', Symbol('key2')] }))
 
 expectType<string>(server.printRoutes())
+
+server.decorate('test', function (x: string) {
+  expectType<FastifyInstance | any>(this)
```

Comment:
If it is `FastifyInstance | any`, I think it would become any at the end?

Reply:
I think you're right. But then the tests are actually useless, aren't they?

Reply:
We can't do `expectType<FastifyRequest>(this)`?

Reply:
For me, the test then fails. I guess the problem is that the type of the decorator value is types as ``any``.
```typescript
  decorate(property: string | symbol, value: any, dependencies?: string[]): FastifyInstance<RawServer, RawRequest, RawReply, Logger>;

```

Reply:
AFAIK, any overload on top of `any` the type will be merged to becomes `any`. So, either change the `any` to something generic without `Function` or this PR should be no actual gain.